spanner/spannertest/db_test.go

```diff
@@ -361,6 +362,83 @@ func TestConcurrentReadInsert(t *testing.T) {
 	}
 }
 
+func TestGeneratedColumn(t *testing.T) {
+	sql := `CREATE TABLE Songwriters (
+		Id INT64 NOT NULL,
+		A INT64,
+		B INT64,
+		C INT64 AS ((2 * A) * B) STORED,
+	) PRIMARY KEY (Id);`
+	var db database
+
+	ddl, err := spansql.ParseDDL("filename", sql)
+	if err != nil {
+		t.Fatalf("%s: Bad DDL", err)
+	}
+	for _, stmt := range ddl.List {
+		if st := db.ApplyDDL(stmt); st.Code() != codes.OK {
+			t.Fatal()
+		}
+	}
+	tx := db.NewTransaction()
+	err = db.Insert(tx, "Songwriters",
+		[]spansql.ID{"Id", "C"},
+		[]*structpb.ListValue{
+			listV(stringV("3"), stringV("3")),
+		})
+	if err == nil || status.Code(err) != codes.InvalidArgument {
+		t.Fatal("Should have failed to insert to generated column")
+	}
+
+	err = db.Insert(tx, "Songwriters",
+		[]spansql.ID{"Id", "A", "B"},
+		[]*structpb.ListValue{
+			listV(stringV("3"), stringV("3"), stringV("3")),
+		})
+	if err != nil {
+		t.Fatalf("Should have succeeded to insert to without generated column: %v", err)
+	}
+
+	var kr keyRangeList
+	iter, err := db.Read("Songwriters", []spansql.ID{"Id", "C"},
+		[]*structpb.ListValue{
+			listV(stringV("3")),
+		}, kr, 0)
+	if err != nil {
+		t.Fatalf("failed to read: %v", err)
+	}
+	rows := slurp(t, iter)
+	// 2 * A * B = 2 * 3 * 3
+	if rows[0][1].(int64) != 18 {
+		t.Fatal("Generated value for C should have been 18")
```

Comment:
nit: use got/want format, so something like `Generated value for C mismatch\n Got: %v\nWant: %v`